File: app/src/main/java/com/ortiz/touchdemo/SwitchImageExampleActivity.java
package com.ortiz.touchdemo;

import android.app.Activity;
import android.os.Bundle;
import android.view.View;
import android.view.View.OnClickListener;

import com.example.touch.R;
import com.ortiz.touchview.TouchImageView;

public class SwitchImageExampleActivity extends Activity {
	
	private TouchImageView image;
	private static int[] images = { R.drawable.nature_1, R.drawable.nature_4, R.drawable.nature_6, R.drawable.nature_7, R.drawable.nature_8 };
	int index = 0;
	
	@Override
	protected void onCreate(Bundle savedInstanceState) {
		super.onCreate(savedInstanceState);
		setContentView(R.layout.activity_switch_image_example);
		image = (TouchImageView) findViewById(R.id.img);
		//
		// Set first image
		//
		if (savedInstanceState != null) {
			index = savedInstanceState.getInt("index");
		}
		image.setImageResource(images[index]);
		//
		// Set next image with each button click
		//
		image.setOnClickListener(new OnClickListener() {
			@Override
			public void onClick(View v) {
				index = (index + 1) % images.length;
				image.setImageResource(images[index]);
			}
		});
	}

	@Override
	protected void onSaveInstanceState(Bundle outState) {
		super.onSaveInstanceState(outState);
		outState.putInt("index", index);
	}
}

File: app/src/main/java/info/touchimage/demo/SwitchImageExampleActivity.kt
package info.touchimage.demo

import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import info.touchimage.demo.databinding.ActivitySwitchImageExampleBinding

class SwitchImageExampleActivity : AppCompatActivity() {

    private var index = 0
    private lateinit var binding: ActivitySwitchImageExampleBinding

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        // https://developer.android.com/topic/libraries/view-binding
        binding = ActivitySwitchImageExampleBinding.inflate(layoutInflater)
        val view = binding.root
        setContentView(view)

        // Set first image
        savedInstanceState?.getInt("index")?.let(this::index::set)
        binding.imageSwitch.setImageResource(images[index])

        // Set next image with each button click
        binding.imageSwitch.setOnClickListener {
            index = (index + 1) % images.size
            binding.imageSwitch.setImageResource(images[index])
        }
    }

    override fun onSaveInstanceState(outState: Bundle) {
        super.onSaveInstanceState(outState)
        outState.putInt("index", index)
    }

    companion object {
        private val images = intArrayOf(R.drawable.nature_1, R.drawable.nature_4, R.drawable.nature_6, R.drawable.nature_7, R.drawable.nature_8)
    }
}
